src/WebJobs.Script/Description/FunctionInvokerBase.cs

```diff
@@ -28,15 +28,14 @@ public abstract class FunctionInvokerBase : IFunctionInvoker, IDisposable
         private IMetricsLogger _metrics;
         private IDisposable _fileChangeSubscription;
 
-        internal FunctionInvokerBase(ScriptHost host, FunctionMetadata functionMetadata, ITraceWriterFactory traceWriterFactory = null)
+        internal FunctionInvokerBase(ScriptHost host, FunctionMetadata functionMetadata)
         {
             Host = host;
             Metadata = functionMetadata;
             _metrics = host.ScriptConfig.HostConfig.GetService<IMetricsLogger>();
 
             // Function file logging is only done conditionally
-            traceWriterFactory = traceWriterFactory ?? new FunctionTraceWriterFactory(functionMetadata.Name, Host.ScriptConfig);
-            TraceWriter traceWriter = traceWriterFactory.Create();
+            TraceWriter traceWriter = host.FunctionTraceWriterFactory.Create(functionMetadata.Name);
```

Comment:
Sounds good. Yeah, we don't need to change this pattern as part of your PRs as this is really out of scope, but these hidden dependencies are often painful.